code/game/objects/items/weapons/RCD.dm

```diff
@@ -580,31 +568,30 @@
 	if(!is_type_in_list(A, allowed_targets))
 		return FALSE
 
-	switch(mode)
-		if(MODE_TURF)
-			. = mode_turf(A, user)
-		if(MODE_AIRLOCK)
-			. = mode_airlock(A, user)
-		if(MODE_DECON)
-			. = mode_decon(A, user)
-		if(MODE_WINDOW)
-			. = mode_window(A, user)
-		else
-			to_chat(user, "ERROR: RCD in MODE: [mode] attempted use by [user]. Send this text #coderbus or an admin.")
-			. = 0
+	for(var/act_type in possible_actions)
+		var/datum/rcd_act/act = new act_type
+		if(act.can_act(A, src, user))
+			. = act.try_act(A, src, user)
+			update_icon(UPDATE_OVERLAYS)
+			SStgui.update_uis(src)
+			return
```

Comment:
```suggestion
	for(var/act_type in possible_actions)
		var/datum/rcd_act/act = new act_type
		if(act.can_act(A, src, user))
			. = act.try_act(A, src, user)
			update_icon(UPDATE_OVERLAYS)
			SStgui.update_uis(src)
			return
```
Please new these objects on init so you don't need to new them every time you attack with an rcd